pkg/proxy/winkernel/proxier.go

```diff
@@ -279,6 +279,9 @@ func (proxier *Proxier) onServiceMapChange(svcPortName *proxy.ServicePortName) {
 
 		klog.V(3).Infof("Updating existing service port %q at %s:%d/%s", svcPortName, svcInfo.ClusterIP(), svcInfo.Port(), svcInfo.Protocol())
 		svcInfo.cleanupAllPolicies(proxier.endpointsMap[*svcPortName])
+		// Delete the endpoint map entry for this service, since the
+		// endpoints are already cleanedup
+		delete(proxier.endpointsMap, *svcPortName)
```

Comment:
This is a  problem. The endpoints might be still there and the Svc might have been updated. We clean up all policies assuming they might be stale. This code is unconditionally deleting all endpoints from the map.

Reply:
I opened a new PR for this issue. #93368